### PR TITLE
chore: Add debounce to search and metrics 

### PR DESCRIPTION
**Pull request by jmoguilevsky**

### CODE REVIEW

File: src/components/Layout/SearchInput.tsx [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
 
 export default function SearchInput(
   props: React.InputHTMLAttributes<HTMLInputElement>
 ) {
+  const { onChange } = props
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    onChange!(e)
+  }
+
+  const debouncedHandleChange = useMemo(() => debounce(handleChange, 350), [])
```
Comment (braianj) [inline src/components/Layout/SearchInput.tsx:16]: WDYT about useCallback?

File: src/components/Place/PlaceCard/PlaceCard.tsx [hunk 2]
```diff
@@ -62,6 +66,18 @@ export default React.memo(function PlaceCard(props: PlaceCardProps) {
     }
   }, [place])
 
+  const handleClickCard = (e: React.MouseEvent<HTMLAnchorElement>) => {
+    e.preventDefault()
+    if (props.search) {
+      track(SegmentPlace.PlaceCardClick, {
+        place: place?.id,
```
Comment (braianj) [inline src/components/Place/PlaceCard/PlaceCard.tsx:73]: I think we use place for the position in the site, like `world_list`, that would be `place_id` and maybe the place should came from props ¿?

File: src/components/Place/PlaceCard/PlaceCard.tsx [hunk 3]
```diff
@@ -73,14 +89,7 @@ export default React.memo(function PlaceCard(props: PlaceCardProps) {
...
     >
-      <a
-        className="place-card__cover"
-        href={href}
-        onClick={(e: React.MouseEvent<HTMLAnchorElement>) => {
-          e.preventDefault()
-          href && navigate(href)
-        }}
-      >
+      <a className="place-card__cover" href={href} onClick={handleClickCard}>
```
Comment (braianj) [inline src/components/Place/PlaceCard/PlaceCard.tsx:92]: There is a component `<Link>` that has tracking embed in it (you can send tracking data as props) and navigate too

File: src/pages/worlds.tsx [hunk 4]
```diff
@@ -78,6 +78,12 @@ export default function WorldsPage() {
     })
 
     if (isSearching) {
+      track(SegmentPlace.WorldsSearch, {
+        resultsCount: placesFetch.total,
+        top10: placesFetch.data.slice(0, 10),
+        search,
+        place: SegmentPlace.Worlds,
```
Comment (braianj) [inline src/pages/worlds.tsx:85]: maybe `SegmentPlace.WorldsSearch`?

**Review by braianj — CHANGES_REQUESTED** [review]

File: src/pages/places.tsx [hunk 5]
```diff
@@ -80,6 +80,13 @@ export default function IndexPage() {
     })
 
     if (isSearching) {
+      track(SegmentPlace.PlacesSearch, {
+        resultsCount: placesFetch.total,
+        top10: placesFetch.data.slice(0, 10),
+        search,
+        place: SegmentPlace.Places,
```
[the remark this answers is not in the document]
Comment (jmoguilevsky) [reply src/pages/places.tsx:87]: The place is places, the event is places search in my opinion

File: src/pages/worlds.tsx (hunk 4 above)
Comment (jmoguilevsky) [reply src/pages/worlds.tsx:85]: Same as places, the place is worlds

**Review by braianj — APPROVED** [review]

**Pull request merged**